src/Components/Admin/EnvironmentEditor/environment.js

```diff
@@ -180,6 +188,33 @@ export default function EnvironmentEditor({
         setScenes(copiedScenes);
     };
 
+    const onDeleteButtonClick = (sceneId) => {
+        setDeleteSceneId(sceneId);
+        setDeleteModalOpen(true);
+    };
+
+    const onDeleteModalCancel = () => {
+        setDeleteSceneId(null);
+        setDeleteModalOpen(false);
+    };
+
+    const onDeleteModalSubmit = async () => {
+        await deleteScene(deleteSceneId, handleError);
+
+        const modifiedScenes = scenes.filter(
+            (scene) => scene.id !== deleteSceneId
+        );
+
+        const newEnvData = environment;
+        newEnvData.scene_ids = modifiedScenes.map((scene) => scene.id);
+        const newEnv = await editScenario(newEnvData, handleError);
```

Comment:
@ahmedhamodi If you look at this logic here, on delete, we actually update the `scene_ids` field on the scenario. Are you planning on changing this so that the `deleteScene` endpoint automatically does this?

Reply:
Yeah I've already implemented that in the PR mentioned above. It goes through all the scenarios and removes the associated scene.

This is better to do from endpoint side because it can handle all the other scenarios impacted by this -- the frontend should not know this information.